src/NativeScript/GlobalObject.mm

```diff
@@ -118,6 +119,27 @@ static void microtaskRunLoopSourcePerformWork(void* context) {
     self->drainMicrotasks();
 }
 
+static void runLoopBeforeWaitingPerformWork(CFRunLoopObserverRef observer, CFRunLoopActivity activity, void* info) {
```

Comment:
The activity is specified when we create the observer so this should be called only on kCFRunLoopBeforeWaiting
https://github.com/NativeScript/ios-runtime/pull/561/files#diff-c94a128c20882442378c1dfa04930835R213